src/app/context/AccountContext.tsx

```diff
@@ -167,9 +182,10 @@ export function AccountProvider({ children }: { children: React.ReactNode }) {
       fiatBalance,
     },
     fetchAccountInfo,
-    loading,
-    balanceLoading,
+    statusLoading,
```

Comment:
`statusLoading` is the original `loading`, it is set in https://github.com/lujakob/lightning-browser-extension/blob/fde9d695c3535252de25abe228eee6e308abf51b/src/app/context/AccountContext.tsx#L156-L157 (only on mount) and used in https://github.com/lujakob/lightning-browser-extension/blob/fde9d695c3535252de25abe228eee6e308abf51b/src/app/router/RequireAuth.tsx#L8-L9

`balanceLoading` is the new one that is used on account change.

I'm not 100% sure anymore, but I think RequireAuth line 8 returns null and nothing is displayed if I used the original, now renamed `statusLoading` for account change. Does it make sense to you?

Reply:
I looked into this when updating the SWR PR - each time you re-open the popup (or any other root page) it will query the background script to get the status which includes whether or not the extension has been unlocked or not (when you close the popup all state there is destroyed). Since the fiat balance requires an API call it needs a separate loading state, but I think that would also be nice to move to SWR.